### PR TITLE
control/controlclient: use lite map request handler to avoid aborting streams

### PR DESCRIPTION
**Pull request by bradfitz**
Previously, any change to endpoints or hostinfo (or hostinfo's
netinfo) would result in the long-running map request HTTP stream
being torn down and restarted, losing all compression context along
with it.

This change makes us instead send a lite map request (OmitPeers: true,
Stream: false) that doesn't subscribe to anything, and then the
coordination server knows to not close other streams for that node
when it recives a lite request.

Fixes tailscale/corp#797

### CODE REVIEW

**Comment by bradfitz** [comment]
This fails the integration tests right now because lite updates don't trigger status updates in the integration test.

The easiest fix is in control: if we get an endpoint-only update, make any existing long-poll connection get a MapResponse update. Sound good, @crawshaw? Then I'll wait on submitting this until that's done & deployed.

File: control/controlclient/auto.go [hunk 1]
```diff
@@ -201,6 +202,48 @@ func (c *Client) Start() {
...
+		c.inLiteMapUpdate = false
+		c.mu.Unlock()
+		if err == nil {
+			c.logf("[v1] successful lite map update in %v", d)
+			return
+		}
+		if ctx.Err() == nil {
+			c.logf("lite map update after %v: %v", d, err)
+		}
+		// Fall back to restarting it heavily, if the
```
Comment (josharian) [inline control/controlclient/auto.go:242]: Sentence fragment

**Review by josharian — COMMENTED** [review]
Seems fine to me, but going to leave for Crawshaw for a second look.

**Comment by bradfitz** [comment]
Waiting to submit until control is deployed with latest changes (including a server-side bug fix) to make this safe to use.

But otherwise this should be good.

**Pull request merged**